src/gui/profilepagewidget.cpp

```diff
@@ -0,0 +1,82 @@
+#include "profilepagewidget.h"
+#include "guiutility.h"
+#include "theme.h"
+
+#include <QPushButton>
+
+namespace OCC {
+
+ProfilePageWidget::ProfilePageWidget(QWidget *parent)
+    : QWidget(parent)
+{
+}
+
+void ProfilePageWidget::setProfileConnector(
+    std::unique_ptr<OcsProfileConnector> profileConnector, const QString &userId)
+{
+    _profileConnector = std::move(profileConnector);
+    _profileConnector->fetchHovercard(userId);
```

Comment:
if we do not check the validity of the pointer, why using one ?
looks to me like a bug waiting to happen

Reply:
Okay, you are right that does not make sense. It's a leftover because I had the `ProfileConnector` base class. I will change it.